src/winrm_client.py

```diff
@@ -0,0 +1,102 @@
+import base64
+import os
+import winrm
+
+
+class WinRMClient:
+    def __init__(self, host, username, password):
+        self.session = winrm.Session(host, auth=(username, password), transport='basic',
+                                     server_cert_validation='ignore')
+
+    def upload_file(self, local_path, remote_path):
+        if not os.path.exists(local_path):
+            raise FileNotFoundError(f"The file {local_path} does not exist.")
+
+        with open(local_path, 'rb') as file:
+            content = file.read()
+        encoded_content = base64.b64encode(content).decode('utf-8')
+
+        # Create directory if it doesn't exist
+        directory = os.path.dirname(remote_path)
+        create_dir_script = f"""
+        if (-not (Test-Path -Path "{directory}")) {{
+            New-Item -Path "{directory}" -ItemType Directory -Force
+            Write-Output "Directory created: {directory}"
+        }} else {{
+            Write-Output "Directory already exists: {directory}"
+        }}
+        """
+        result = self.session.run_ps(create_dir_script)
+        dir_creation_output = result.std_out.decode().strip()
+        dir_creation_error = result.std_err.decode().strip()
+
+        print(f"Directory creation result: {dir_creation_output}")
+        if dir_creation_error:
+            print(f"Directory creation error: {dir_creation_error}")
+
+        # Verify the directory was created
+        dir_check_script = f"Test-Path -Path '{directory}'"
+        dir_exists_result = self.execute_command(dir_check_script).strip().lower()
+        print(f"Directory exists check result: {dir_exists_result}")
+        if dir_exists_result != "true":
+            raise FileNotFoundError(f"Directory {directory} does not exist after creation.")
+
+        # Upload the file
+        upload_script = f"""
```

Comment:
Please create module for powershell scripts and move all PS scripts templated to that file